src/io.cr

```diff
@@ -173,7 +173,7 @@ abstract class IO
   # io << "Crystal"
   # io.to_s # => "1-Crystal"
   # ```
-  def <<(obj) : self
+  def <<(obj : _) : self
```

Comment:
I didn't even know this was an option until I saw other usages of it in this file. Is this the equivalent of `Object`?

Reply:
`_` has a special meaning, standing in for "anything". See https://crystal-lang.org/reference/1.16/syntax_and_semantics/type_grammar.html#underscore

But yes, effectively, that makes it pretty much equivalent to `Object` (but shorter).

Reply:
This is not allowed:

```crystal
def foo(x : Object)
end

foo([1].as(Enumerable(Int32)))
```

because `Enumerable(Int32) <= Object` does not hold, but `_` will make it work.

Reply:
IMO this dummy type restriction just adds noise to the code without serving any particular purpose. More than anything I reckon it will only make ppl confused re: the `_` usage as the type restriction.

Reply:
The type restriction `_` _explicitly_ documents that the type can be _anything_. That's different from _no_ type restriction, which just means the information is missing. That's a significant difference.

Reply:
@straight-shoota So what's the difference then?

Reply:
If there's no type restriction, it _could_ mean the type is in fact unrestricted. But that's rarely the case.
But most often, there are limitations on what types are accepted, it's just not expressed explicitly.
There's no information, and thus no way to tell what it means.

A type restriction to `_` makes it clear that the type is supposed to be anything.